src/entry-client.js

```diff
@@ -1,6 +1,10 @@
-import { find, isFunction, isString } from 'lodash-es';
+import { get, find, isFunction, isString } from 'lodash-es';
 
+// We only ignore route updates between when routing between the same entries
+// in the routing table (i.e., Catch-All->Catch-All or PDP->PDP).  Never ignore
+// route updates between routing table entries.
 const shouldIgnoreRouteUpdate = (c, args) => (
+    get(args, 'from.name') === get(args, 'route.name') &&
     isFunction(c.shouldIgnoreRouteUpdate) &&
     c.shouldIgnoreRouteUpdate(args) === true
```

Comment:
On a separate note - I did find myself wondering if this should be reversed.  Right now, the default is to run the middlewares/fetchData unless a route opts out.  But from what we've seen the normal behavior is to _not_ run and only category pages opt-in at the moment.  So maybe it would make more sense to move to something like `shouldProcessRouteUpdate`?  Right now I think there's probably a number of routes that do not implement this method, but only work fine because they do not run into routing updates with changed query params mid route.

Reply:
I like the _ignore by default_ option unless you opt in. Personally I don't think we should we re-running anything unless we specifically opt-in like you said. It also feels more impactful that you have to choose to turn it on for a given reason, especially if the majority of our routes aren't using `shouldIgnoreRouteUpdate` at the moment.

Reply:
ok - since that would be a breaking change - I think I'll lump that in with the upcoming npm dependency bumps which will likely require a major change as well.  So we can get this in the 4.6.1 version and then switch the logic in v5

Reply:
Yep, thanks Broph 👍